lib/daun/rugged_daun.rb

```diff
@@ -81,7 +82,21 @@ def fetch_refs
       # Prune is not supported by rugged! Deleting all remote refs and re-fetch
       delete_all_remote_branches
       delete_all_tags
-      @repository.remotes['origin'].fetch
```

Comment:
It's best if this logic can be moved out of this class. We can pass credentials object as an optional parameter to the `checkout` method. The object creation can then be done by `cli.rb`. This will allow daun to support more use cases in the future while keeping `rugged_daun.rb` clean, such as:

- Custom location of privatekey and publickey
- SSH key passphrase
- `Credentials::SshKeyFromAgent` instead of `Credentials::SshKey`

All of the complexity above can be abstracted out by passing in command line argument.